homeassistant/helpers/template.py

```diff
@@ -832,6 +832,43 @@ def async_render_with_possible_json_value(
 
         return self._parse_result(render_result)
 
+    @callback
+    def async_render_as_value_template(
+        self,
+        variables: dict[str, Any],
+        error_value: Any = _SENTINEL,
+    ) -> Any:
```

Comment:
I'm not sure about adding this very specific method here. 

Maybe make a subclass of `Template` in `trigger_template_entity.py`:

```py
class ValueTemplate(Template):
    @classmethod
    def from_template(cls, template: Template):
        return cls(template.template, template.hass)

    @callback
    def async_render_as_value_template(
        self,
        variables: dict[str, Any],
        error_value: Any = _SENTINEL,
    ) -> Any:
        ...
```

Then, in the schemas you can just do something like this:

```py
vol.Optional(CONF_VALUE_TEMPLATE): vol.All(cv.template, ValueTemplate.from_template)
```

Reply:
My concern with this method is that these changes only affect trigger entities, and if we change the validation it will update to all template style entities.  If you're ok with this, I'll go this route.